### PR TITLE
Add Sonata music player (MPD client)

### PR DESCRIPTION
**Pull request by rvl**
This is my favourite MPD (Music Player Daemon) client and one of its python dependencies.

Cheers.

cc: @multani @Mic92

### CODE REVIEW

**Comment by multani** [comment]
Cool :)
If there are things that should be merged in Sonata, feel free to open PR with patches, I don't have much time/interest to work on Sonata at the moment, but I can certainly merge things if needed.

File: pkgs/applications/audio/sonata/default.nix [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  postPatch = ''
+    # Fix python wheel version
+    substituteInPlace setup.py \
+      --replace "sonata.svnversion.VERSION" "\"${version}\""
+
+    # Remove "Local MPD" tab which doesn't work.
+    sed -i '/localmpd/d' sonata/consts.py
+
+    # Add a fix for github issue #41
```
Comment (multani) [inline pkgs/applications/audio/sonata/default.nix]: Sonata:#41 has been merged, do you still require this?

**Comment by rvl** [comment]
Hi @multani, the first 2 patches are more due to the "unique" nature of NixOS than any problem with Sonata.

The 3rd, a fix for multani/sonata#41 relates to v1.7a2 which is earlier than the merge.

We can take any revision from GitHub, but I think a tagged version is best. Is it possible to make a new tag of a suitable revision for packaging? Ideally that tag would have the fix for multani/sonata#41 as well as the gtk.TextView segfault fix.

Cheers

**Comment by multani** [comment]
On 01/31/2016 03:32 PM, Rodney Lorrimar wrote:
> Is it possible to make a new tag of a suitable revision for
> packaging? Ideally that tag would have the fix for multani/sonata#41
> <https://github.com/multani/sonata/pull/41> as well as the gtk.TextView
> segfault fix.

I just released a new version which should have these fixes. Thanks for 
the packaging!

**Comment by rvl** [comment]
Thanks @multani, package is updated to your new version.

**Comment by fpletz** [comment]
Thanks to you all! This was like so many stuff also on my TODO-list. ;)

There seems to be one issue, though. When starting sonata, this message is displayed:
```
GLib-GIO-Message: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
```

This usually means that dconf is not available but sonata does seem to persist the configuration so I'm not sure if we need to fix that. If you use `wrapGAppsHook` most of the environment variables you wrap manually will be wrapped automatically like GIO_EXTRA_MODULES for dconf.

**Comment by rvl** [comment]
Hi @fpletz, I don't see that message. I have tested under Gnome 3 and xmonad (within a gnome-session). Which desktop environment are you using? In the Sonata code, I can't find any references to dconf or gsettings, and the only reference to GIO is in a drag&drop callback.

**Comment by multani** [comment]
Sonata doesn't use dconf nor GSettings directly, but it does use some of the "high-level" feature of Gtk which may use these libraries, although I can't see this in the documentation (I'm thinking of ``GtkApplication`` especially).

**Comment by colemickens** [comment]
I put this on my "To Package" list yesterday and today see the PR. Thanks!

**Comment by rvl** [comment]
This is rebased on latest master, and now using `wrapGAppsHook`, thanks @fpletz.

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -11457,6 +11457,31 @@ in modules // {
     };
   };
 
+  mpd2 = buildPythonPackage rec {
+    name = "python-mpd2-${version}";
+    version = "0.5.5";
+
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/p/python-mpd2/${name}.tar.bz2";
+      md5 = "50ba833dbfda7fca30088ed1deabe351";
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you use a sha256 here

Comment (rvl) [inline pkgs/top-level/python-packages.nix]: OK fixed thanks @FRidh

**Comment by fpletz** [comment]
Thanks & sorry for the long wait. :beers:

**Comment by rvl** [comment]
No prob, thanks for the reviews and merge @fpletz @FRidh @multani.

**Pull request merged**